src/java/org/apache/cassandra/index/sai/disk/SearchableIndex.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import org.apache.cassandra.db.DecoratedKey;
+import org.apache.cassandra.db.PartitionPosition;
+import org.apache.cassandra.db.virtual.SimpleDataSet;
+import org.apache.cassandra.dht.AbstractBounds;
+import org.apache.cassandra.index.sai.SSTableQueryContext;
+import org.apache.cassandra.index.sai.plan.Expression;
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+import org.apache.cassandra.index.sai.utils.RangeIterator;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+
+/**
+ * This is used to abstract the index search between on-disk versions.
+ * Callers to this interface should be unaware of the on-disk version for
+ * the index.
+ *
+ * It is responsible for supplying metadata about the on-disk index. This is
+ * used during query time to help coordinate queries and is also returned
+ * by the virtual tables.
+ */
+public abstract class SearchableIndex implements Closeable
+{
+    public abstract long indexFileCacheSize();
+
+    public abstract long getRowCount();
+
+    public abstract long minSSTableRowId();
+
+    public abstract long maxSSTableRowId();
+
+    public abstract ByteBuffer minTerm();
```

Comment:
I'm not sure what you are getting at here. The terms are passed around as `ByteBuffer`. There is no usage of these terms that uses `ByteComparable`.

Reply:
The encoding format of a `ByteBuffer` is unclear.

Reply:
`ByteComparable` doesn't imply any encoding format. `ByteBuffer` is used because it is used throughout the rest of Cassandra for data values and can be reused in comparisons. `ByteComparable` is only used in the trie and is effectively single use.